### PR TITLE
Docker support

### PR DESCRIPTION
**Pull request by LeandroGuillen**
Implements docker support, including a docker compose file and a README.

It implements @jmcanterafonseca changes in harness tests that he developed in #1057.

### CODE REVIEW

File: docker/README.md [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+	  docker run -d --name orion1 --link mongodb:mongodb -p 1026:1026 orion -dbhost mongodb
+
+As a result of this command, there is a context broker listening on port 1026 on localhost. Try to see if it works now with
+
+	curl localhost:1026/version
+
+A few points to consider:
+
+* The name `orion1` can be anything and doesn't have to be related to the name given to the docker image in the previous section.
+* `--link mongodb:mongodb` assumes there is a docker container running a MongoDB image in your system, whose name is `mongodb`. In case you need one type `docker run --name mongodb -d mongo`.
```
Comment (fgalan) [inline docker/README.md]: How do you ensue that the MongoDB you are using is a 2.6 one?

Comment (LeandroGuillen) [inline docker/README.md]: I don't, I use latest

File: docker/README.md [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+* In `-p 1026:1026` the first value represents the port to listen in on localhost. If you wanted to run a second context broker on your machine you should change this value to something else, for example `-p 1027:1026`.
+* Anything after the name of the container image (in this case `orion`) is interpreted as a parameter for the Orion Context Broker. In this case we are telling the broker where the MongoDB host is, represented by the name of our other MongoDB container. Take a look at the [documentation](../.../.../doc/admin/cli.md) for other command-line options.
+
```
Comment (fgalan) [inline docker/README.md]: The doc should also include a section explining how to run the test harness againts the just started Orion docker

File: docker/Dockerfile [hunk 3]
```diff
@@ -0,0 +1,15 @@
+FROM centos:centos6
```
Comment (fgalan) [inline docker/Dockerfile]: Maybe this could be more precise and point to a CentOS 6 particular version? E.g. 6.3, 6.4, 6.5.

File: docker/Dockerfile [hunk 4]
```diff
@@ -0,0 +1,15 @@
...
+MAINTAINER FIWARE Orion Context Broker Team. Telefónica I+D
+
+# Setup FIWARE repo
+RUN echo -e "[testbed-fi-ware]\n\
+name=Fiware Repository\n\
+baseurl=http://repositories.testbed.fi-ware.org/repo/rpm/x86_64/\n\
```
Comment (fgalan) [inline docker/Dockerfile]: fi-ware.org -> fiware.org

File: docker/Dockerfile [hunk 5]
```diff
@@ -0,0 +1,15 @@
+FROM centos:centos6
+MAINTAINER FIWARE Orion Context Broker Team. Telefónica I+D
+
+# Setup FIWARE repo
+RUN echo -e "[testbed-fi-ware]\n\
```
Comment (fgalan) [inline docker/Dockerfile]: testbed-fi-ware -> testbed-fiware

File: docker/Dockerfile [hunk 6]
```diff
@@ -0,0 +1,15 @@
...
+name=Fiware Repository\n\
+baseurl=http://repositories.testbed.fi-ware.org/repo/rpm/x86_64/\n\
+gpgcheck=0\n\
+enabled=1\n" > /etc/yum.repos.d/testbed-fi-ware.repo
+
+# Install Orion
+RUN yum -y install contextBroker
+
+EXPOSE 1026
+ENTRYPOINT ["/usr/bin/contextBroker", "-fg", "-t 0-255"]
```
Comment (fgalan) [inline docker/Dockerfile]: It is not a good idea to enable `-t 0-255`, the log is very verbose.

File: docker/Dockerfile (hunk 3 above)
Comment (LeandroGuillen) [inline docker/Dockerfile]: You can only tell it to run CentOS 6 (latest branch) or CentOS 6.6 specifically 8https://registry.hub.docker.com/_/centos/).

File: docker/README.md (hunk 1 above)
Comment (fgalan) [inline docker/README.md]: Yes, 2.6 should be enforced.

File: docker/Dockerfile (hunk 3 above)
Comment (fgalan) [inline docker/Dockerfile]: I think is not a good idea depending on "moving pictures", so I guess that CentOS 6.6 should be used here.

File: docker/README.md (hunk 2 above)
Comment (fgalan) [inline docker/README.md]: This section is not mandatory for this PR (as it will be re-worked when we develop the final solution based in 2 dockers).

**Comment by jmcanterafonseca** [comment]
I have a general comment on this docker. I assume that the environment on which Orion was compiled is the same as the one on which it is executed. Otherwise, we cannot guarantee library compatibility at runtime, right?

**Comment by LeandroGuillen** [comment]
This packaged Orion works on CentOS, the supported OS. And the dockerfile pulls from CentOS 6.6. So yes, it is guaranteed to work.

File: docker/README.md [hunk 7]
```diff
@@ -0,0 +1,48 @@
...
+As a result of this command, there is a context broker listening on port 1026 on localhost. Try to see if it works now with
+
+	curl localhost:1026/version
+
+A few points to consider:
+
+* The name `orion1` can be anything and doesn't have to be related to the name given to the docker image in the previous section.
+* `--link mongodb:mongodb` assumes there is a docker container running a MongoDB image in your system, whose name is `mongodb`. In case you need one type `docker run --name mongodb -d mongo`.
+* In `-p 1026:1026` the first value represents the port to listen in on localhost. If you wanted to run a second context broker on your machine you should change this value to something else, for example `-p 1027:1026`.
+* Anything after the name of the container image (in this case `orion`) is interpreted as a parameter for the Orion Context Broker. In this case we are telling the broker where the MongoDB host is, represented by the name of our other MongoDB container. Take a look at the [documentation](../.../.../doc/admin/cli.md) for other command-line options.
```
Comment (fgalan) [inline docker/README.md]: Self-comment: add `httpsPrepare.sh` script to contextBroker-tests RPM. @LeandroGuillen any other script you are missing in the RPM?

Comment (LeandroGuillen) [inline docker/README.md]: I would leave that to another PR, actually

Comment (fgalan) [inline docker/README.md]: Good point. I would use a new PR to fix the RPM. In the meanwhile, if you find you need other files in the RPM, use this comment thread to provide that feedback.

File: docker/Dockerfile (hunk 3 above)
Comment (jmcanterafonseca) [inline docker/Dockerfile]: if we go for only one official Docker per GE, maybe we are not going to need a docker folder and we might put the Dockerfile at the root of the repo, WDYT?

File: docker/README.md (hunk 1 above)
Comment (LeandroGuillen) [inline docker/README.md]: fixed in 5bd9c16

File: docker/README.md (hunk 2 above)
Comment (LeandroGuillen) [inline docker/README.md]: I have done the test harness in a very manual way. Leaving this for another PR.

NTC

File: docker/Dockerfile (hunk 3 above)
Comment (LeandroGuillen) [inline docker/Dockerfile]: @jmcanterafonseca We still have 3 files, a README, a dockerfile, and a docker-compose. I leave it as is now, but if you feel it would be better at the root of the project I'll move it.

NTC?

File: docker/Dockerfile (hunk 4 above)
Comment (LeandroGuillen) [inline docker/Dockerfile]: Fixed in d865a27

File: docker/Dockerfile (hunk 5 above)
Comment (LeandroGuillen) [inline docker/Dockerfile]: Fixed in d865a27

File: docker/Dockerfile (hunk 6 above)
Comment (LeandroGuillen) [inline docker/Dockerfile]: Fixed in 5e6a53f

File: docker/README.md (hunk 7 above)
Comment (LeandroGuillen) [inline docker/README.md]: New issue created #1083

**Comment by fgalan** [comment]
LGTM

File: docker/Dockerfile (hunk 3 above)
Comment (jmcanterafonseca) [inline docker/Dockerfile]: Perfect

**Comment by fgalan** [comment]
@jmcanterafonseca has provided his LGTM out of band (by skype)
